components/scream/src/share/util/scream_vertical_interpolation.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef SCREAM_VERTICAL_INTERPOLATION_HPP
+#define SCREAM_VERTICAL_INTERPOLATION_HPP
+
+#include "share/scream_types.hpp"
+
+#include "ekat/util/ekat_lin_interp.hpp"
+#include "ekat/ekat_pack_utils.hpp"
+#include "ekat/kokkos/ekat_subview_utils.hpp"
+
+namespace scream {
+
+//
+// ------- Types --------
+//
+
+template <typename S>
+using SmallPack = ekat::Pack<S,SCREAM_SMALL_PACK_SIZE>;
+using Spack = SmallPack<Real>;
+
+template <typename S>
+using SmallPack = ekat::Pack<S,SCREAM_SMALL_PACK_SIZE>;
+using Sbool = SmallPack<bool>;
+
+using Smask = ekat::Mask<Spack::n>;
+
+using KT = KokkosTypes<DefaultDevice>;
+using ExeSpace = typename KT::ExeSpace;
+using ESU = ekat::ExeSpaceUtils<ExeSpace>;
+using LIV = ekat::LinInterp<Real,Spack::n>;
+
+ 
+template <typename S>
+using view_1d = typename KT::template view_1d<S>;
+template <typename S>
+using view_2d = typename KT::template view_2d<S>;
+
+const Real masked_val = -9.99e17;
```

Comment:
This seems like an arbitrary number. If the goal is to put some invalid number, prefer using `ekat::ScalarTraits<Real>::invalid()`.

Reply:
In development this number was discussed with Peter and Aaron as being preferable to a nan when being able to analyze offline. Also, capability to be able to set it to a different number was requested, which is currently in the code. I can revisit this with them.

Reply:
Yeah, @bartgol - the idea was that we didn't want to have to store both 3d mask and 3d data for each variable in the netcdf output. it sounds like `ekat::ScalarTraits<Real>::invalid()` might be preferable, but I wonder if python analysis code would get confused by this? I guess my question is - what is this invalid() thing? Is it an actual number like -999? Or will it write NaN in the netcdf file? Or is it something else?

Reply:
This was updated to be: -std::numeric_limits<Real>::max() since a nan is not preferred